diff --git a/board/variscite/mx6var_som/mx6var_som.c b/board/variscite/mx6var_som/mx6var_som.c
index 9557ca1b45..5903a54d0d 100644
--- a/board/variscite/mx6var_som/mx6var_som.c
+++ b/board/variscite/mx6var_som/mx6var_som.c
@@ -129,118 +129,6 @@ static inline bool is_dart_board(void)
 	return is_cpu_pop_packaged();
 }
 
-
-#define PUS_100K_DOWN_PAD_CTRL (PAD_CTL_PUS_100K_DOWN | \
-	PAD_CTL_SPEED_HIGH | PAD_CTL_DSE_40ohm |			\
-	PAD_CTL_SRE_FAST)
-#define PUS_100K_UP_PAD_CTRL (PAD_CTL_PUS_100K_UP | \
-	PAD_CTL_SPEED_HIGH | PAD_CTL_DSE_40ohm |			\
-	PAD_CTL_SRE_FAST)
-
-static iomux_v3_cfg_t const gpio4_21_gpio3_21_en[] = {
-	IOMUX_PADS(PAD_DISP0_DAT0__GPIO4_IO21	| MUX_PAD_CTRL(PUS_100K_DOWN_PAD_CTRL)),
-	IOMUX_PADS(PAD_EIM_D21__GPIO3_IO21		| MUX_PAD_CTRL(PUS_100K_UP_PAD_CTRL)),
-};
-static iomux_v3_cfg_t const gpio4_21_gpio3_21_dis[] = {
-	IOMUX_PADS(PAD_DISP0_DAT0__GPIO4_IO21	| MUX_PAD_CTRL(NO_PAD_CTRL)),
-	IOMUX_PADS(PAD_EIM_D21__GPIO3_IO21		| MUX_PAD_CTRL(NO_PAD_CTRL)),
-};
-
-enum iris2_ident {
-	UNKNOWN = 0,
-	IRIS2,
-	NIGHTCRAWLER,
-};
-
-/* List of GPIOs that need to be set to INPUT */
-static iomux_v3_cfg_t const gpio_nightcrawler_dis[] = {
-	IOMUX_PADS(PAD_ENET_MDC__GPIO1_IO31	| MUX_PAD_CTRL(NO_PAD_CTRL)), /* DART_IO_0 10K PD */
-	IOMUX_PADS(PAD_ENET_MDIO__GPIO1_IO22	| MUX_PAD_CTRL(NO_PAD_CTRL)), /* DART_IO_1 10K PD */
-	IOMUX_PADS(PAD_DISP0_DAT2__GPIO4_IO23	| MUX_PAD_CTRL(NO_PAD_CTRL)), /* DART_IO_2 10K PD */
-	IOMUX_PADS(PAD_DISP0_DAT15__GPIO5_IO09	| MUX_PAD_CTRL(NO_PAD_CTRL)), /* DART_IO_3 10K PD */
-	IOMUX_PADS(PAD_DISP0_DAT16__GPIO5_IO10	| MUX_PAD_CTRL(NO_PAD_CTRL)), /* SILVUS_SHUTDOWN 10K PD */
-};
-
-/*
- * Returns 1 if the carrier board is Iris2, 2 if carrier is Nightcrawler
- *  (and the SOM is DART-MX6)
- */
-static inline int get_iris2_ident(void)
-{
-	int ident = UNKNOWN;
-
-	SETUP_IOMUX_PADS(gpio4_21_gpio3_21_en);
-
-	gpio_request(IMX_GPIO_NR(4, 21), "");
-	gpio_request(IMX_GPIO_NR(3, 21), "");
-	gpio_direction_input(IMX_GPIO_NR(4, 21));
-	gpio_direction_input(IMX_GPIO_NR(3, 21));
-
-	lsb = gpio_get_value(IMX_GPIO_NR(4, 21));
-	msb = gpio_get_value(IMX_GPIO_NR(3, 21));
-
-	if(msb == 1){
-		if(lsb == 1) ret = IRIS2;
-		else ret = NIGHTCRAWLER;
-	}
-
-	SETUP_IOMUX_PADS(gpio4_21_gpio3_21_dis);
-	return ret;
-}
-
-enum iris2_rev {
-	IRIS2_R0 = 0,
-	IRIS2_R1,
-	IRIS2_R2,
-	IRIS2_R3,
-	IRIS2_NONE,
-};
-
-static iomux_v3_cfg_t const gpio4_29_30_en[] = {
-	IOMUX_PADS(PAD_DISP0_DAT8__GPIO4_IO29	| MUX_PAD_CTRL(PUS_100K_UP_PAD_CTRL)),
-	IOMUX_PADS(PAD_DISP0_DAT9__GPIO4_IO30	| MUX_PAD_CTRL(PUS_100K_UP_PAD_CTRL)),
-};
-static iomux_v3_cfg_t const gpio4_29_30_dis[] = {
-	IOMUX_PADS(PAD_DISP0_DAT8__GPIO4_IO29	| MUX_PAD_CTRL(NO_PAD_CTRL)),
-	IOMUX_PADS(PAD_DISP0_DAT9__GPIO4_IO30	| MUX_PAD_CTRL(NO_PAD_CTRL)),
-};
-
-/**
- * Gets the iris2/nightcrawler revision
- * GPIO4[30] 	GPIO4[29] 	Revision
- * HIGH 		HIGH 		R0
- * HIGH 		LOW 		R1
- * LOW 			HIGH 		R2
- * LOW 			LOW 		R3
- * 
- */
-static inline int get_iris2_rev(void)
-{
-	int lsb, msb, board;
-
-	SETUP_IOMUX_PADS(gpio4_29_30_en);
-	
-	gpio_request(IMX_GPIO_NR(4, 29), "");
-	gpio_request(IMX_GPIO_NR(4, 30), "");
-	gpio_direction_input(IMX_GPIO_NR(4, 29));
-	gpio_direction_input(IMX_GPIO_NR(4, 30));
-
-	lsb = gpio_get_value(IMX_GPIO_NR(4, 29));
-	msb = gpio_get_value(IMX_GPIO_NR(4, 30));
-
-	if(msb == 1){
-		if(lsb == 1) board = IRIS2_R0;
-		else board = IRIS2_R1;
-	}
-	else{
-		if(lsb == 1) board = IRIS2_R2;
-		else board = IRIS2_R3;
-	}
-
-	SETUP_IOMUX_PADS(gpio4_29_30_dis);
-	return board;
-}
-
 /*
  * Returns true iff the carrier board is VAR-MX6CustomBoard
  */
@@ -1437,52 +1325,18 @@ int power_init_board(void)
 #ifndef CONFIG_SPL_BUILD
 int board_late_init(void)
 {
-#ifdef CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
-	int ident = 0;
-#endif
 #ifdef CONFIG_ENV_IS_IN_MMC
 	mmc_late_init();
 #endif
 	print_emmc_size();
 
 #ifdef CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
-	ident = get_iris2_ident();
-	if (ident != UNKNOWN) {
-		int revision = get_iris2_rev();
-
-		if (ident == IRIS2) {
-			/*SETUP_IOMUX_PADS(gpio_iris2_dis);*/
-			if (revision == IRIS2_R0)
-				setenv("board_name", "IRIS2_R0");
-			else if (revision == IRIS2_R1)
-				setenv("board_name", "IRIS2_R1");
-			else if (revision == IRIS2_R2)
-				setenv("board_name", "IRIS2_R2");
-			else if (revision == IRIS2_R3)
-				setenv("board_name", "IRIS2_R3");
-			else
-				setenv("board_name", "DT6CUSTOM");
-		} else if (ident == NIGHTCRAWLER) {
-			SETUP_IOMUX_PADS(gpio_nightcrawler_dis);
-			if (revision == IRIS2_R0)
-				setenv("board_name", "NIGHTCRAWLER_R0");
-			else if (revision == IRIS2_R1)
-				setenv("board_name", "NIGHTCRAWLER_R1");
-			else if (revision == IRIS2_R2)
-				setenv("board_name", "NIGHTCRAWLER_R2");
-			else if (revision == IRIS2_R3)
-				setenv("board_name", "NIGHTCRAWLER_R3");
-			else
-				setenv("board_name", "DT6CUSTOM");
-		}
-	} else {
-		if (is_dart_board())
-			setenv("board_name", "DT6CUSTOM");
-		else if (is_solo_custom_board())
-			setenv("board_name", "SOLOCUSTOM");
-		else
-			setenv("board_name", "MX6CUSTOM");
-	}
+	if (is_dart_board())
+		setenv("board_name", "DT6CUSTOM");
+	else if (is_solo_custom_board())
+		setenv("board_name", "SOLOCUSTOM");
+	else
+		setenv("board_name", "MX6CUSTOM");
 
 	if (is_som_solo())
 		setenv("board_som", "SOM-SOLO");
diff --git a/include/configs/mx6var_som.h b/include/configs/mx6var_som.h
index 9196d6c814..1db7afb4d1 100644
--- a/include/configs/mx6var_som.h
+++ b/include/configs/mx6var_som.h
@@ -243,18 +243,6 @@
 		"fi;\0" \
 	"findfdt="\
 		"if test $fdt_file = undefined; then " \
-			"if test $board_name = IRIS2_R0; then " \
-				"setenv fdt_file imx6q-iris2-R0.dtb; " \
-			"fi; " \
-			"if test $board_name = IRIS2_R1; then " \
-				"setenv fdt_file imx6q-iris2-R1.dtb; " \
-			"fi; " \
-			"if test $board_name = IRIS2_R2; then " \
-				"setenv fdt_file imx6q-iris2-R2.dtb; " \
-			"fi; " \
-			"if test $board_name = IRIS2_R3; then " \
-				"setenv fdt_file imx6q-iris2-R3.dtb; " \
-			"fi; " \
 			"if test $board_name = DT6CUSTOM && test $board_rev = MX6Q; then " \
 				"setenv fdt_file imx6q-var-dart.dtb; " \
 			"fi; " \
